src/utils/wallet.ts

```diff
@@ -35,3 +63,34 @@ export const isSafe = async (wallet: ConnectedWallet): Promise<boolean | null> =
     return false
   }
 }
+
+const isHardwareWallet = (wallet: ConnectedWallet): boolean => {
+  return [WALLET_KEYS.LEDGER, WALLET_KEYS.TREZOR, WALLET_KEYS.KEYSTONE].includes(
+    wallet.label.toUpperCase() as WALLET_KEYS,
+  )
+}
+
+export const switchWalletChain = async (
+  onboard: OnboardAPI,
+  connectedWallet: ConnectedWallet,
+  chainId: number,
+): Promise<void> => {
+  if (!isHardwareWallet(connectedWallet)) {
+    await onboard.setChain({ wallet: connectedWallet.label, chainId: hexValue(chainId) })
+    return
```

Comment:
```suggestion
    return onboard.setChain({ wallet: connectedWallet.label, chainId: hexValue(chainId) })
```

Reply:
`setChain` returns a `boolean` but this function doesn't have a `return` value. We can either return a `boolean` (in every other place as well) or leave it as is. What do you think?

Reply:
It was more like a nit since we don't do anything after the call we usually don't need to await but lets keep it as it is.